OpenKinect/libfreenect2
# Add VA-API JPEG decoder

This adds VA-API support for Intel GPUs under Linux.

Combined with OpenCL, the performance given Intel i7-4600U/HD Graphics 4400:
````
[OpenCLDepthPacketProcessor] avg. time: 14.1288ms -> ~70.7774Hz
[VaapiJpegRgbPacketProcessor] avg. time: 5.07581ms -> ~197.013Hz
````

JPEG decoding consumes less than 10% of a single core.

This pull request can't be merged as-is. It depends on #221. After those two are merged, dependencies in this PR will be cleaned out.

* v2:

Use memory mapped buffers in input and output to avoid extra memory copy. This requires modifications to input and output buffer structures `Frame` and `DoubleBuffer`.

* **Test instructions**:

To test this branch, you must have i965-va-driver installed. To avoid degraded color decoding, you can update libva and i965-va-driver to 1.5.0 by temporarily adding vivid to /etc/apt/sources.list, and then update only i965-va-driver and libva-dev.

You may see this warning on Ubuntu 14.04:
```
[VaapiJpegRgbPacketProcessor::initializeVaapi] warning: YUV444 not supported by libva, chroma will be halved
```
This is because libva on Ubuntu 14.04 is not new enough. Follow instructions above to update libva and i965-va-driver.
